pkg/rancher-desktop/main/commandServer/settingsValidator.ts

```diff
@@ -65,13 +65,15 @@ export default class SettingsValidator {
   allowedTransientSettings: TransientSettingsValidationMap | null = null;
   synonymsTable: settingsLike|null = null;
   lockedSettings: LockedSettingsType = { };
+  isFatal = false;
```

Comment:
I find it rather weird that this is a property on the validator object instead of a return value for the `validateSettings()` method, just like `errors`. I expected usage to be like this:

```typescript
const [needToUpdate, errors, isFatal] = settingsValidator.validateSettings(cfg, newSettings, lockedFields);
```

Because `isFatal` is closely related to values returned in `errors` and not a global state of the object.

You can see this even in the tests you wrote because they look like this:

```typescript
expect({ needToUpdate, errors, isFatal: subject.isFatal }).toEqual({
...
```

Reply:
ok, I moved it to a return value (which is ignored most of the time outside the unit tests)